### PR TITLE
Zendesk go live notes

### PR DESCRIPTION
**Pull request by QuanNguyen-CDS**
# Summary | Résumé
[Trello](https://trello.com/c/AH8gsNqx/465-update-deal-in-zensales-with-go-live-request-notes)
feat: hooked in go live request into the already existing contact request rest API call
refactor: removed all the test and implementation code for freshdesk.create_ticket and the rest call `/<uuid:user_id>/support-email`
feat: updated zendesk sell with `create_note`, `search for deal`, and `create go live request`

# Test instructions | Instructions pour tester la modification
Validated that zendesk go live request populates zendesk sell
ran unit tests: `test_contact_request`, `test_zendesk_sell`, `test_freskdesk`, `test_rest`

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet
Note there a matching change on notification-admin which will come very shortly.

# Reviewer checklist | Liste de vérification du réviseur

This is a suggested checklist of questions reviewers might ask during their
review | Voici une suggestion de liste de vérification comprenant des questions
que les réviseurs pourraient poser pendant leur examen :

- [ ] Does this meet a user need? | Est-ce que ça répond à un besoin utilisateur?
- [ ] Is it accessible? | Est-ce que c’est accessible?
- [ ] Is it translated between both offical languages? | Est-ce dans les deux
      langues officielles?
- [ ] Is the code maintainable? | Est-ce que le code peut être maintenu?
- [ ] Have you tested it? | L’avez-vous testé?
- [ ] Are there automated tests? | Y a-t-il des tests automatisés?
- [ ] Does this cause automated test coverage to drop? | Est-ce que ça entraîne
      une baisse de la quantité de code couvert par les tests automatisés?
- [ ] Does this break existing functionality? | Est-ce que ça brise une
      fonctionnalité existante?
- [ ] Should this be split into smaller PRs to decrease change risk? | Est-ce
      que ça devrait être divisé en de plus petites demandes de tirage (« pull
      requests ») afin de réduire le risque lié aux modifications?
- [ ] Does this change the privacy policy? | Est-ce que ça entraîne une
      modification de la politique de confidentialité?
- [ ] Does this introduce any security concerns? | Est-ce que ça introduit des
      préoccupations liées à la sécurité?
- [ ] Does this significantly alter performance? | Est-ce que ça modifie de
      façon importante la performance?
- [ ] What is the risk level of using added dependencies? | Quel est le degré de
      risque d’utiliser des dépendances ajoutées?
- [ ] Should any documentation be updated as a result of this? (i.e. README
      setup, etc.) | Faudra-t-il mettre à jour la documentation à la suite de ce
      changement (fichier README, etc.)?

### CODE REVIEW

File: app/clients/zendesk_sell.py [hunk 1]
```diff
@@ -111,6 +117,31 @@ def _generate_lead_conversion_data(lead_id: int):
             }
         }
 
+    @staticmethod
+    def _generate_note_data(resource_type: str, resource_id: str, content: str) -> Dict[str, str]:
```
Comment (AntoineAugusti) [inline app/clients/zendesk_sell.py:121]:
```suggestion
    def _generate_note_data(resource_type: NoteResourceType, resource_id: str, content: str) -> Dict[str, str]:
```
I believe

File: app/user/rest.py [hunk 2]
```diff
@@ -476,9 +457,12 @@ def send_contact_request(user_id):
...
         pass
 
-    # FIXME: Since this is a POC, do not include the status code for zendesk as a part of the return code
     try:
-        ZenDeskSell().send_contact_request(contact)
+        if contact.is_go_live_request():
+            service = dao_fetch_service_by_id(contact.service_id)
+            ZenDeskSell().send_go_live_request(service, user, contact)
+        else:
+            ZenDeskSell().send_contact_request(contact)
```
Comment (AntoineAugusti) [inline app/user/rest.py:465]: Can you add a test covering sending a go live request here?

File: app/clients/zendesk_sell.py [hunk 3]
```diff
@@ -249,6 +296,32 @@ def upsert_deal(self, contact_id: int, service: Service, stage_id: int) -> Optio
...
+                    resource_type: NoteResourceType,
+                    resource_id: str,
+                    contact: ContactRequest) -> Optional[str]:
+
+        # The API and field definitions are defined here:
+        # https://developers.getbase.com/docs/rest/reference/notes
+        resp, e = self._send_request(
+            method='POST',
+            relative_url='/v2/notes',
+            data=json.dumps(
```
Comment (AntoineAugusti) [inline app/clients/zendesk_sell.py:309]: I believe you can just pass a `json` arg to `requests`, don't know if your private method accepts that

File: app/clients/freshdesk.py [hunk 4]
```diff
@@ -31,14 +31,25 @@ def _generate_description(self):
                 f'- main use case: {self.contact.main_use_case}',
                 f'- main use case details: {self.contact.main_use_case_details}',
             ])
+        elif self.contact.is_go_live_request():
+            message = '<br>'.join([
```
Comment (AntoineAugusti) [inline app/clients/freshdesk.py:35]: I see that we use 2 `<br>` for demo requests, why is it different? What is the right thing?

Comment (QuanNguyen-CDS) [reply app/clients/freshdesk.py:35]: you are right, the question is why is the demo request different than the request to go live.
If you remember we moved the message generation from notification-admin into notification-api.
Functionally the message should be identical from the previous implementation going to freshdesk for the `demo` and `request to go live`. If we want to consider change the format of the message this is a change in functional logic at the freshdesk level

File: app/clients/zendesk_sell.py (hunk 1 above)
Comment (QuanNguyen-CDS) [reply app/clients/zendesk_sell.py:121]: nice catch!

File: app/clients/freshdesk.py (hunk 4 above)
Comment (AntoineAugusti) [reply app/clients/freshdesk.py:35]: Sounds good

**Review by AntoineAugusti — APPROVED** [review]
LGTM! Nice work.

I'll let you decide if you want to do more refactoring before merging otherwise it's all good

**Pull request merged**